db/background_mgr_attachment_migration.go

```diff
@@ -0,0 +1,351 @@
+//  Copyright 2024-Present Couchbase, Inc.
+//
+//  Use of this software is governed by the Business Source License included
+//  in the file licenses/BSL-Couchbase.txt.  As of the Change Date specified
+//  in that file, in accordance with the Business Source License, use of this
+//  software will be governed by the Apache License, Version 2.0, included in
+//  the file licenses/APL2.txt.
+
+package db
+
+import (
+	"context"
+	"fmt"
+	"slices"
+	"strings"
+	"sync"
+
+	sgbucket "github.com/couchbase/sg-bucket"
+	"github.com/couchbase/sync_gateway/base"
+	"github.com/google/uuid"
+)
+
+type AttachmentMigrationManager struct {
+	DocsProcessed base.AtomicInt
+	DocsChanged   base.AtomicInt
+	MigrationID   string
+	CollectionIDs []uint32
+	lock          sync.RWMutex
+}
+
+var _ BackgroundManagerProcessI = &AttachmentMigrationManager{}
+
+func NewAttachmentMigrationManager(metadataStore base.DataStore, metaKeys *base.MetadataKeys) *BackgroundManager {
+	return &BackgroundManager{
+		name:    "attachment_migration",
+		Process: &AttachmentMigrationManager{},
+		clusterAwareOptions: &ClusterAwareBackgroundManagerOptions{
+			metadataStore: metadataStore,
+			metaKeys:      metaKeys,
+			processSuffix: "attachment_migration",
+		},
+		terminator: base.NewSafeTerminator(),
+	}
+}
+
+func (a *AttachmentMigrationManager) Init(ctx context.Context, options map[string]interface{}, clusterStatus []byte) error {
+	newRunInit := func() error {
+		uniqueUUID, err := uuid.NewRandom()
+		if err != nil {
+			return err
+		}
+
+		a.MigrationID = uniqueUUID.String()
+		base.InfofCtx(ctx, base.KeyAll, "Attachment Migration: Starting new migration run with migration ID: %q", a.MigrationID)
```

Comment:
```suggestion
		base.InfofCtx(ctx, base.KeyAll, "Attachment Migration: Starting new migration run with migration ID: %s", a.MigrationID)
```

We log this as `%s` below and I think consistency when reading logs can be helpful.